main.cpp

```diff
@@ -159,47 +177,58 @@ struct AquariumTank
         float maxSalinity = 0.5f;
         float age = 0.f;
         float maxAge = 3.f;
+        ~Fish(){}
 
-        //
-        bool canLiveInThisWater(float waterPH, float waterSalinity);
+        bool canLiveInThisWater(float waterPH, float waterSalinity)
+        {
+            if((waterPH >= 6.f && waterPH <= 8.f) && (waterSalinity < maxSalinity))
+                return true;
+            return false;
+        }
         void ageFish(float amountOfTime);
         bool isStillAlive(float waterPH, float waterSalinity)
         {
             return (canLiveInThisWater(waterPH, waterSalinity) && (age < maxAge));
         }
     };
-    // 1) add fish
-    void addFish(Fish fish, int Quantity);
-    // 2) add water
+    void addFish(Fish* fish, int quantity);
     void addWater(float amountOfWater);
-    // 3) adjust pH
     void adjustPH(float phAdjustment);
 };
 
-/*
-Thing 2) Museum
-5 properties:
-    1) the number of exhibits (int)
-    2) the number of employees (int)
-    3) the total cost of utilities used each month (float)
-    4) the monthly retail income amount (float)
-    5) the amount of government subsidy recieved each month (double)
-3 things it can do:
-    1) charge visitor
-    2) add or remove employees
-    3) lobby politicians for more government funding
- */
+void AquariumTank::addFish(AquariumTank::Fish *fish, int quantity)
+{
+    if( fish->canLiveInThisWater(pHLevel, 0.f))
+    {
+        //todo: add quantity of fish to a list
+        fishLivingIn += quantity;
+    }
+}
+
+void AquariumTank::addWater(float amountOfWater)
+{
+    float currentWaterAmount = waterCapacity * currentWaterLevel;
+    currentWaterAmount += amountOfWater;
+    if(currentWaterAmount > waterCapacity)
+        currentWaterAmount = waterCapacity; // maybe add a "spilled function to alert someone"
+    currentWaterLevel = currentWaterAmount / waterCapacity;
+}
+
+void AquariumTank::adjustPH(float phAdjustment)
+{
+    pHLevel += phAdjustment;
+    if(pHLevel > 14.f)
+        pHLevel = 14.f;
+    if(pHLevel < 0.f)
+        pHLevel = 0.f;
+}
```

Comment:
This will all be much more readable if you add a blank line after each `if()` body
```
if( thing )
    expression();
    
moreExpressions();
```
- [ ] revise